src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  constructor() {
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super();
+    this.name = name;
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+
+  bite(herb) {
+    if (!herb.hidden && !(herb instanceof Carnivore)) {
```

Comment:
[FUNCTION NAMING] - The `bite` method should check if the `herb` argument is an instance of `Herbivore` and not hidden. Currently, the check for `!(herb instanceof Carnivore)` is incorrect and should be replaced with `herb instanceof Herbivore` to meet the task requirements.